File: blog-automation/scripts/collect.mjs
// 웹페이지 본문 수집 → 마크다운 저장 (Playwright + Cheerio + Turndown)
// 사용: node scripts/collect.mjs https://example.com
import { chromium } from 'playwright';
import * as cheerio from 'cheerio';
import TurndownService from 'turndown';
import fs from 'node:fs/promises';

const url = process.argv[2];
if (!url) { console.error('사용법: node scripts/collect.mjs <URL>'); process.exit(1); }

const browser = await chromium.launch();
const page = await browser.newPage({
  userAgent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 Chrome/124.0 Safari/537.36',
});
await page.goto(url, { waitUntil: 'domcontentloaded', timeout: 30000 });
const html = await page.content();
await browser.close();

const $ = cheerio.load(html);
$('script, style, nav, footer, header, aside, iframe, noscript').remove();
const title = $('meta[property="og:title"]').attr('content') || $('title').text().trim();
const desc = $('meta[property="og:description"]').attr('content')
  || $('meta[name="description"]').attr('content') || '';
const body = $('article').html() || $('main').html() || $('body').html();

const md = new TurndownService({ headingStyle: 'atx', codeBlockStyle: 'fenced' }).turndown(body || '');
const slug = (title || 'untitled').replace(/[\/:*?"<>|]/g, '').slice(0, 50).trim() || 'untitled';
const outPath = `data/${slug}.md`;

await fs.mkdir('data', { recursive: true });
await fs.writeFile(outPath, `# ${title}\n\n> ${desc}\n>\n> 출처: ${url}\n> 수집: ${new Date().toISOString()}\n\n---\n\n${md}\n`);
console.log(`수집 완료 → ${outPath} (${md.length.toLocaleString()}자)`);

File: blog-automation/scripts/collect.py
"""정적 페이지 다건 수집 → 엑셀 저장 (requests + BeautifulSoup + pandas)

사용: uv run scripts/collect.py https://example.com https://example.org
"""
import sys
from datetime import datetime

# Windows 콘솔에서 한글 출력이 깨지지 않도록
if hasattr(sys.stdout, "reconfigure"):
    sys.stdout.reconfigure(encoding="utf-8", errors="replace")

import pandas as pd
import requests
from bs4 import BeautifulSoup

HEADERS = {"User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) Chrome/124.0 Safari/537.36"}


def scrape(url: str) -> dict:
    res = requests.get(url, headers=HEADERS, timeout=15)
    res.raise_for_status()
    soup = BeautifulSoup(res.text, "lxml")

    def meta(prop: str, attr: str = "property") -> str:
        tag = soup.find("meta", attrs={attr: prop})
        return tag.get("content", "") if tag else ""

    for junk in soup(["script", "style", "nav", "footer", "header", "aside"]):
        junk.decompose()

    return {
        "url": url,
        "title": meta("og:title") or (soup.title.string.strip() if soup.title else ""),
        "description": meta("og:description") or meta("description", "name"),
        "image": meta("og:image"),
        "text": " ".join(soup.get_text(" ", strip=True).split())[:2000],
        "collected_at": datetime.now().isoformat(timespec="seconds"),
    }


def main() -> None:
    urls = sys.argv[1:]
    if not urls:
        print("사용법: uv run scripts/collect.py <URL> [URL ...]")
        raise SystemExit(1)

    rows = []
    for url in urls:
        try:
            rows.append(scrape(url))
            print(f"  OK  {url}")
        except Exception as exc:  # 한 건 실패가 전체를 막지 않도록
            print(f"  실패 {url} — {exc}")

    if not rows:
        print("수집된 항목이 없습니다.")
        return

    out = "data/수집결과.xlsx"
    pd.DataFrame(rows).to_excel(out, index=False)
    print(f"수집 완료 → {out} ({len(rows)}건)")


if __name__ == "__main__":
    main()
